core/src/main/java/org/apache/iceberg/RewriteTablePathUtil.java

```diff
@@ -126,8 +126,7 @@ public static TableMetadata replacePaths(
         metadata.snapshotLog(),
         metadataLogEntries,
         metadata.refs(),
-        // TODO: update statistic file paths
-        metadata.statisticsFiles(),
+        updatePathInStatisticsFiles(metadata.statisticsFiles(), sourcePrefix, targetPrefix),
         metadata.partitionStatisticsFiles(),
```

Comment:
we may still need a `todo` comment for `partitionStatisticsFiles` :-)

Reply:
sounds good, I added precondition check and TODO comments for partition stats